File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> food) {
    string ans = "", rev="";

    for(int i=1;i<food.size();i++){
        for(int j=0;j<(int)food[i]/2;j++){
            ans.push_back(48+i);
        }
    }

    rev = ans;
    reverse(rev.begin(), rev.end());
    return ans+"0"+rev;
}

File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] foods) {
        int[] cnt = new int[10];

        for(int i=1;i<foods.length;i++){
            cnt[i] = foods[i]/2;
        }


        StringBuilder sb = new StringBuilder();
        for(int i=1;i<10;i++){
            if(cnt[i] == 0) continue;

            for(int j=0;j<cnt[i];j++)
                sb.append(i);
        }

        return sb.toString() + "0" + sb.reverse().toString();
    }
}